### PR TITLE
Updated and split the documentation.

### PR DESCRIPTION
**Pull request by marcushanses**

### CODE REVIEW

**Comment by marcushanses** [comment]
`No fix available.`
image

File: integration.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+```
+The binding in the kv-namespaces and the name of the durable_objects bindings can be whatever you want.
+
+### Opening the DB
+Lets say we want to open a transactional database that can store users with an interface User.
+```ts
+import { User } from "../model/User"
+import * as gracely from "gracely"
+
+type Layout = { collection: { user: User } }
```
Comment (nilssonemma) [inline integration.md:32]: How do we decide how the Layout type should look like? Is it always an object with a collection prop or is that just an example?

**Review by nilssonemma — CHANGES_REQUESTED** [review]

File: integration.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+}
+```
+Here the the dependency `gracely` is used to generate errors. Note that the `Storage.environment.archive` & `Storage.environment.BufferBackend` needs to correspond to the bindings defined in wrangler.toml. The initialize function needs to be called with the environment by the worker before utilizing the database. The types for the configuration can be found in the [configuraion folder](Database/Configuration).
+
+Now the database can be used by the worker as follows:
+```ts
+import { Storage } from "./Storage"
+import { User } from "../model/User"
+
+export async function store(user: User, organization: string): Promise<User> {
```
[the remark this answers is not in the document]
Comment (marcushanses) [reply integration.md:60]: This is just an example, like user is also just a random type.
Should I be more clear that this is an example?

File: integration.md (hunk 1 above)
Comment (marcushanses) [reply integration.md:32]: The layout can be different, should I link to the type in the documentation?

File: developer.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+- The Archive is able to store, load and remove documents, loading can be done using the Selection type or an id of the document.
+- The Collection utilizes a Durable Object abstraction called Buffer and an Archive for long term storage. The Buffer ensures consistency and transactionality while writing. The buffer stores documents for a short timespan and writes documents to the Archive once deemed stale by the Archivist. The buffer can contain several durable objects to increase the number of simultaneous read and writes, these durable objects will use a key-shard, the shards are generated by the ids of the documents.
+
+The Database can be partitioned to store data in different layers. The key for the documents are comprised of `typeName/doc/partition_1/partition_2/.../partition_n/dateTime/documentID` where n is a natural number. The Buffer will divide the data in different durable objects using `typeName/partition_1/partition_2/.../partition_n/shard` as key.
+
+![f](https://user-images.githubusercontent.com/79835961/183652283-8623068b-e8e2-47ab-ad63-691375a9e26b.png)
+
+_figure 1: The architecture of the database._
+
+## Indicies
```
Comment (Hlebanja) [inline developer.md:16]: indices

File: developer.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+There exists 3 types of keys in the Database, `id`, `changed` and `doc`; each adds functionality to the database.
+The definitions below are based on how they look in the `Archive`.
+
+### doc
+The doc-index contains the document the user wants to store/load/update.
+The structure of the doc-key makes it possible to list certain a partition and query it on created time with a date range.
```
Comment (Hlebanja) [inline developer.md:22]: certain a -> a certain

**Review by Hlebanja — CHANGES_REQUESTED** [review]
Small fixes; overall really well written ;)

File: developer.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+	                                 key_n`
+}
+```
+Where the truncate function truncates the changed date of the document to minutes to save all keys of documents changed within that minute, the document is the value the user wants to store in the database, the key_n are on the form of key defined in `doc` and n is a natural number.
+
+## Shard
+The shard is calculated from the first two characters in the id of the document. Since the id of the document is assumed to be base 64 the first two characters will contain 12 bits of information. By converting the characters to a Uint8Array and then truncating, such that only as many bits needed to describe as many shards configured is used, followed by a conversion back to base 64, a surjective mapping is created from id -> shard.
+
+## Cursor
+The Cursor's content should be obscured from the end user, this is done by base 64 encode the data. Encoding is enough since we're not dealing with security, rather ensuring the end user doesn't consider the Cursor to be mutable.
```
Comment (Hlebanja) [inline developer.md:61]: by base 64 encoding of the data
or
by base 64 encoding the data

**Review by Hlebanja — CHANGES_REQUESTED** [review]

File: integration.md (hunk 1 above)
Comment (nilssonemma) [reply integration.md:32]: Yes that would be good. Makes it easier to know how to write the layout :)

**Review by nilssonemma — APPROVED** [review]

**Review by Hlebanja — APPROVED** [review]

**Pull request merged**